reo/src/data_manager.py

```diff
@@ -1216,6 +1321,40 @@ def finalize(self):
         non_cooling_electric_load = [self.load.load_list[ts] - cooling_load[ts] / self.cooling_load.chiller_cop for ts in range(len(self.load.load_list))]
         non_cooling_electric_load_bau = [self.load.bau_load_list[ts] - cooling_load[ts] / self.cooling_load.chiller_cop for ts in range(len(self.load.bau_load_list))]
 
+        # GHP parameters for REopt model
+        require_ghp_purchase, force_ghp_bau = 0, 0
+        ghp_installed_cost, ghp_installed_cost_bau = [], []
+        ghp_om_cost_year_one, ghp_om_cost_year_one_bau = [], []
+        ghp_heating_thermal_load_served_kw, ghp_heating_thermal_load_served_kw_bau = [], [[]]
+        ghp_cooling_thermal_load_served_kw, ghp_cooling_thermal_load_served_kw_bau = [], [[]]
+        ghp_electric_consumption_kw, ghp_electric_consumption_kw_bau = [], [[]]
+        if len(self.ghp_option_list) > 0:
+            require_ghp_purchase = self.ghp_option_list[0].require_ghp_purchase  # This does not change with the number of options
+            for i, option in enumerate(self.ghp_option_list):
+                ghp_cap_cost_slope, ghp_cap_cost_x, ghp_cap_cost_yint, ghp_n_segments = self._get_REopt_cost_curve(techs=["ghp_option_list[{}]".format(i)])
+                ghp_size_tons = option.heatpump_capacity_tons
+                if ghp_size_tons <= ghp_cap_cost_x[1]:
+                    n = 0
+                elif ghp_size_tons > ghp_cap_cost_x[-1]:
+                    n = ghp_n_segments - 1
+                else:
+                    for n in range(ghp_n_segments):
+                        if (ghp_size_tons > ghp_cap_cost_x[n]) and (ghp_size_tons <= ghp_cap_cost_x[n+1]):
+                            break
+                ghp_installed_cost.append(ghp_cap_cost_yint[n] + ghp_size_tons * ghp_cap_cost_slope[n])
+                ghp_om_cost_year_one.append(option.om_cost_year_one)
+                if self.boiler is not None:
+                    heating_thermal_load = np.array(heating_load) * self.boiler_efficiency
+                else:
+                    heating_thermal_load = np.array(heating_load) * 0.8
+                ghp_heating_thermal_load_served_kw.append(list(np.minimum(option.heating_thermal_load_served_kw, heating_thermal_load)))
```

Comment:
@Bill-Becker I thought that GHP could only meet space heating (or was it only DHW?): the code appears to be setting the `ghp_heating_thermal_load_served_kw` to the minimum of an input and the _sum_ of the space heating and DHW.

Reply:
@NLaws Yes, whether GHP serves space heating (default) or the whole heating load (non-default) is set in scenario.py based on `GHP.can_serve_dhw`. This np.minimum is for rounding differences from the Julia GHPGHX.jl module that could result in GHP serve **slightly more** than the **total** heating load (talking about very small fractions which would make the problem infeasible from the thermal energy balance constraint).